src/main/java/io/javalin/core/util/JettyServerUtil.kt

```diff
@@ -100,10 +98,20 @@ object JettyServerUtil {
         return (server.connectors[0] as ServerConnector).localPort
     }
 
-    private fun attachHandlersToTail(userHandler: Handler?, handlerList: HandlerList): HandlerWrapper {
+    private fun attachHandlersToTail(userHandler: Handler?, handlerList: HandlerList): Handler {
+        fun wrap(handlerList: HandlerList, handlerWrapper: HandlerWrapper): Handler {
+            return HandlerWrapper().apply {
+                handler = handlerList
+                insertHandler(handlerWrapper)
+            }.handler
+        }
+
+        if (userHandler is HandlerCollection) {
+            userHandler.addHandler(wrap(handlerList, HandlerWrapper()))
+            return userHandler
+        }
         val handlerWrapper = (userHandler ?: HandlerWrapper()) as HandlerWrapper
-        HandlerWrapper().apply { handler = handlerList }.insertHandler(handlerWrapper)
-        return handlerWrapper
+        return wrap(handlerList, handlerWrapper)
```

Comment:
I had some trouble understanding this, so I checked it out locally and rewrote it. Since the change wants to introduce a special case for allowing a custom `HandlerCollection`, I gathered the new functionality in the `if`, and left the old functionality untouched:

```kotlin
private fun attachHandlersToTail(userHandler: Handler?, handlerList: HandlerList): Handler {
    if (userHandler is HandlerCollection) {
        userHandler.addHandler(handlerList)
        return userHandler
    }
    val handlerWrapper = (userHandler ?: HandlerWrapper()) as HandlerWrapper
    HandlerWrapper().apply { handler = handlerList }.insertHandler(handlerWrapper)
    return handlerWrapper
}
```

Tests pass. Does it do what it's supposed to do?

Reply:
Could be simplified with a `when`: 

```kotlin
private fun attachJavalinHandlers(userHandler: Handler?, javalinHandlers: HandlerList) = when {
    userHandler is HandlerCollection -> userHandler.apply { addHandler(javalinHandlers) }
    userHandler is HandlerWrapper -> HandlerWrapper().apply {
        handler = javalinHandlers
        insertHandler(userHandler)
    }
    else -> HandlerWrapper().apply { handler = javalinHandlers } // no custom handlers set
}
```

Reply:
Could also catch unidentified handlers:

```kotlin
private fun attachJavalinHandlers(userHandler: Handler?, javalinHandlers: HandlerList) = when {
    userHandler == null -> HandlerWrapper().apply { handler = javalinHandlers } // no custom handlers set
    userHandler is HandlerCollection -> userHandler.apply { addHandler(javalinHandlers) }
    userHandler is HandlerWrapper -> HandlerWrapper().apply {
        handler = javalinHandlers
        insertHandler(userHandler)
    }
    else -> throw IllegalStateException("Server has unidentified handler attached to it")
}
```